src/include/compat.h

```diff
@@ -13,10 +13,57 @@
 #define CEPH_COMPAT_H
 
 #if defined(__FreeBSD__)
+
+/* Make sure that ENODATA is defined in the correct way */
+#ifndef ENODATA
 #define	ENODATA	ENOATTR
+#else
+#if (ENODATA == 9919)
+// #warning ENODATA already defined to be 9919, redefining to fix
```

Comment:
> I think you are refering to the text in the discussion, not the actual commit?

yes. thanks for the explanation. now i have a vague understanding now =) what is the "a compile time check" exactly? could you document this as a comment here?